### ISSUE TITLE
lsp-fsharp fails to start up

### ISSUE THREAD

**Issue by ezemtsov** [emacs-lsp/lsp-mode#1508]
LSP version: Since lsp-mode-20200306.626

Steps to reproduce:
1. Start any minimal F# project
2. Lsp fails to start with following error:

LSP :: Newtonsoft.Json.JsonSerializationException: Cannot deserialize the current JSON object (e.g. {"name":"value"}) into type 'System.Nullable`1[System.Boolean]' because the type requires a JSON primitive value (e.g. string, number, boolean, null) to deserialize correctly.
To fix this error either change the JSON to a JSON primitive value (e.g. string, number, boolean, null) or change the deserialized type so that it is a normal .NET type (e.g. not a primitive type like integer, not a collection type like an array or List<T>) that can be deserialized from a JSON object. JsonObjectAttribute can also be added to the type to force it to deserialize from a JSON object.

Related issue:
https://github.com/fsharp/FsAutoComplete/issues/567

**Comment by yyoncho** [comment]
This is server side bug:

```
export interface PublishDiagnosticsClientCapabilities {
	/**
	 * Whether the clients accepts diagnostics with related information.
	 */
	relatedInformation?: boolean;

	/**
	 * Client supports the tag property to provide meta data about a diagnostic.
	 * Clients supporting tags have to handle unknown tags gracefully.
	 *
	 * @since 3.15.0
	 */
	tagSupport?: {
		/**
		 * The tags supported by the client.
		 */
		valueSet: DiagnosticTag[];
	};

	/**
	 * Whether the client interprets the version property of the
	 * `textDocument/publishDiagnostics` notification's parameter.
	 *
	 * @since 3.15.0
	 */
	versionSupport?: boolean;
}
```

As you can see the tagSupport is not boolean but a list.

The workaround on the our side is to comment this line: https://github.com/emacs-lsp/lsp-mode/blob/master/lsp-mode.el#L2960

**Comment by mpardalos** [comment]
@yyoncho The workaround doesn't seem to work for me. Has anyone else tried it?

**Comment by ezemtsov** [comment]
@mpardalos the fix indeed helps.

Another way to solve it is to use lsp-mode version before supporting diagnostic tags. This is the last commit before regression was introduced:
https://github.com/emacs-lsp/lsp-mode/commit/3dd5ced8065132fbc05587bf3f2f0150283030f2

Fix on FSAC side was already merged: https://github.com/fsharp/FsAutoComplete/pull/574
We are waiting for new release of fsautocomplete, after it's done everything is expected to "just work".

**Comment by ezemtsov** [comment]
@yyoncho the bug is fixed on FSAC side with 0.41.1 release: 
https://github.com/fsharp/FsAutoComplete/pull/574

lsp-mode still fails with same issue, should we reopen the issue?

**Comment by yyoncho** [comment]
If the server does not handle the spec correctly then yes.
